server/src/main/java/org/elasticsearch/index/mapper/ParseContext.java

```diff
@@ -319,11 +315,8 @@ public void addIgnoredField(String field) {
 
         private final Set<String> ignoredFields = new HashSet<>();
 
-        public InternalParseContext(IndexSettings indexSettings, DocumentMapperParser docMapperParser, DocumentMapper docMapper,
-                                    SourceToParse source, XContentParser parser) {
-            this.indexSettings = indexSettings;
+        public InternalParseContext(DocumentMapper docMapper, SourceToParse source, XContentParser parser) {
```

Comment:
I am considering making this class package private as a follow-up, its only public because its used in percolator tests.